src/main/java/com/google/sps/servlets/TasksServlet.java

```diff
@@ -45,26 +42,25 @@
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // get Google credential object. Ensure it is valid - otherwise return an error to client
-    Credential googleCredential = AuthenticationUtility.getGoogleCredential(request);
-    if (googleCredential == null) {
-      response.sendError(403, AuthenticationUtility.ERROR_403);
-      return;
-    }
+    // Load and verify google credential
+    super.doGet(request, response);
 
-    // Get tasks from Google Tasks
-    Tasks tasksService = TasksUtility.getTasksService(googleCredential);
-    List<TaskList> taskLists = TasksUtility.listTaskLists(tasksService);
-    List<Task> tasks = new ArrayList<>();
-    for (TaskList taskList : taskLists) {
-      tasks.addAll(TasksUtility.listTasks(tasksService, taskList));
-    }
+    // Credential is null if user is not authenticated.
+    if (googleCredential != null) {
```

Comment:
Same as above - do you want to return an error of some sort to the client if the credential is null like before? I think it would be nice since you're already checking for it.

Reply:
Thanks :)

Reply:
If `googleCredential` should never be null, please throw when it is so we can quickly notice something is wrong. ([Fail fast principle](https://www.martinfowler.com/ieeeSoftware/failFast.pdf).)

Reply:
Oops - I forgot to update this servlet once I made changes to the authenticatedHttpServlet. 

This method is called on the condition that the googleCredential is not null, so this check is redundant. Replacing with assertion